### ISSUE TITLE
Public Key size from DNSKEY 

### ISSUE THREAD

**Issue by nirajsorathiya101** [miekg/dns#1550]
Hello,

I am trying to find out Key size in bits of public key present in DNSKEY record. Unfortunately, I didn't find any method to calculate key size. 

Other option that I thought was to extract modulus from DNSKEY public key. But could not find useful method to  extract modulus as well in  crypto RSA package https://pkg.go.dev/crypto/rsa@go1.22.1#PublicKey .

I would really appreciate if anyone can help me. 

Best ,
Niraj

**Comment by nirajsorathiya101** [comment]
In addition, keytag function  https://pkg.go.dev/github.com/miekg/dns#DNSKEY.KeyTag  looks helpful
Is there any documentation that can help to understand  keytag function ?

**Comment by miekg** [comment]
See RFC 4034 for the keytag

**Comment by nirajsorathiya101** [comment]
@miekg  Thanks for your answer. I referred RFC 4034. Public key format seem to be  in separate RFC. 

Is there  public key format for DNSKEY  algorithm 8 RSA/SHA256 available ?   

I came across https://www.rfc-editor.org/rfc/rfc5702.html#section-6.1  but I don't know how they created DNSKEY record. I little bit understood from keytag function that I need wire format of public key which use algorithm  8 and then which might help to extract modulus from public key and hence the key size.

Best regards,
Niraj
